main.py

```diff
@@ -15,26 +30,138 @@ def print_board(board):
+    # iterate through all ships in botShips
+    for ship in botShips:
+        # before placing ships ensure there is room for the ship after choosing the random coordinate
+        room = False
+        while room == False:
+            # choose random coordinate
+            x = random.randint(0, board_size - 1)
+            y = random.randint(0, board_size - 1)
+
+            # is there room for the ship below the coordinate?
+            if x + ship.hitPoints - 1 < board_size:
+                tempX = x
+                for i in range(ship.hitPoints):
+                    # check if there's an existing ship here
+                    if board[tempX][y] == empty_space:
+                        room = True
+                    else:
+                        room = False
+                        break
+                    tempX+=1
+                if room == True:
+                    tempX = x
+                    # place ship
+                    for i in range(ship.hitPoints):
+                        board[tempX][y] = ship.letter
+                        tempX+=1
+
+            # is there room for the ship to the left of the coordinate?
+            elif y - ship.hitPoints - 1 >= 0:
+                tempY = y
+                for i in range(ship.hitPoints):
+                    if board[x][y] == empty_space:
+                        room = True
+                    else:
+                        room = False
+                        break
+                    tempY-=1
+                if room == True:
+                    tempY = y
+                    for i in range(ship.hitPoints):
+                        board[x][tempY] = ship.letter
+                        tempY-=1
+
+            # is there room for the ship above the coordinate?
+            elif x - ship.hitPoints - 1 >= 0:
+                tempX = x
+                for i in range(ship.hitPoints):
+                    if board[tempX][y] == empty_space:
+                        room = True
+                    else:
+                        room = False
+                        break
+                    tempX-=1
+                if room == True:
+                    tempX = x
+                    for i in range(ship.hitPoints):
+                        board[tempX][y] = ship.letter
+                        tempX-=1
+
+            # is there room for the ship to the right of the coordinate?
+            elif y + ship.hitPoints - 1 < board_size:
+                tempY = y
+                for i in range(ship.hitPoints):
+                    if board[x][tempY] == empty_space:
+                        room = True
+                    else:
+                        room = False
+                        break
+                    y+=1
+                if room == True:
+                    tempY = y
+                    for i in range(ship.hitPoints):
+                        board[x][tempY] = ship.letter
+                        tempY+=1
+
+def createShips():
```

Comment:
much prefer this use case. cleaner, shorter and eliminates the need for magic numbers

`SHIP_TYPES = [
    ("Carrier", 5, "C"),
    ("Battleship", 4, "B"),
    ("Cruiser", 3, "C"),
    ("Submarine", 3, "S"),
    ("Destroyer", 2, "D"),
]

def createShips():
    for ship_class, hit_points, letter in SHIP_TYPES:
        playerShips.append(Ship(ship_class, hit_points, letter))
        botShips.append(Ship(ship_class, hit_points, letter))
`